corehq/apps/case_importer/tracking/models.py

```diff
@@ -0,0 +1,41 @@
+from django.db import models
+from dimagi.ext import jsonobject
+from dimagi.utils.decorators.memoized import memoized
+from soil.util import get_task
+
+
+class CaseUploadRecord(models.Model):
+    domain = models.CharField(max_length=256)
+    created = models.DateTimeField(auto_now_add=True)
+
+    upload_id = models.UUIDField(unique=True)
+    task_id = models.UUIDField(unique=True)
+    couch_user_id = models.CharField(max_length=256)
+    case_type = models.CharField(max_length=256)
+
+    @property
+    @memoized
+    def task(self):
+        return get_task(self.task_id)
+
+
+class CaseUploadJSON(jsonobject.JsonObject):
```

Comment:
I had two objectives in writing this, though I'm not too tied to it and it's definitely more verbose than the alternatives.

1. have an explicitly-defined read-only model for `get_case_uploads` to return, rather than a queryset which lets the caller do all sorts of things to the underlying SQL tables.
2. have a well-defined schema for the JSON object to be passed on for use in javascript

(2) ended up not working out since I later realized I need to do some further transformations before passing it on for use in javascript, but 1 is still a pro. It's definitely adds some redundancy, so I may scrap it anyway for the simpler option, or keep some aspect of it but simplify it somehow.